### PR TITLE
bumper, Add githubApi functions to open PR

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR implements PR creation in the bumper script.

- **bumper, Add githubApi functions to open PR**
This commit implements the functions that create
a proper PR using githubApi.

**Special notes for your reviewer**:
depends on #619 to be merged, relevant commit is the last one

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
An alternative to using allowedList: running the bump script on a CNAO clone
@qinqon what do you think?

File: tools/bumper/component_commands.go
Comment (qinqon) [inline tools/bumper/component_commands.go:519]: we can use short version of commit id.

**Review by alonSadan — COMMENTED** [review]
Thank you Ram. all looks good. I just got a bit confused with the log for loop, so consider using some helpers there.

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:237]: Done

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:227]: since githubInterface.GetRef already an interface funtion (and not a direct all to githubApi), I can just change it's name to `getBrnachRef` like your suggest. is that ok?

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:231]: Done

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go]: also changed `githubInterface` function names to be private in a separate commit

**Review by alonSadan — COMMENTED** [review]
sorry for the helpers nag:)
one last suggestion

File: tools/bumper/cnao_repo_commands.go
Comment (alonSadan) [reply tools/bumper/cnao_repo_commands.go]: yep, sounds good. maybe there is one more thing we can be done here. see https://github.com/kubevirt/cluster-network-addons-operator/pull/509#discussion_r507686112

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:230]: Done

**Comment by alonSadan** [comment]
Thanks @RamLavi 
/lgtm

**Comment by RamLavi** [comment]
/hold
until #619 is merged

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [inline tools/bumper/cnao_repo_commands.go:225]: `s/branchIdx/branchOpenAttempts/g`

Do we really need to retry ?

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [inline tools/bumper/cnao_repo_commands.go:230]: We don't retry if this fails ?

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [inline tools/bumper/cnao_repo_commands.go:235]: Add some sleep to give some air to the network or whatever is failing.

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [inline tools/bumper/cnao_repo_commands.go:242]: This function is not using the receiver

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [inline tools/bumper/cnao_repo_commands.go:243]: We cannot compare with proper error from githubInterface instead of string at Errors ? 

https://blog.golang.org/go1.13-errors, look for `Examining errors with Is and As`

**Review by qinqon — CHANGES_REQUESTED** [review]

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:225]: since it's pseudo random it'll get the same name, but it should be fixed with seed. 
Done

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:230]: there are 3 cases here:
1. branch is not yet opened, so create it
2. branch already created. we want to avoid this, so we use random string. but if it does happen we don't want to continue.
3. the case you asked about, where there's a different error. one reason I can think of is the  github is down.. in that case I think we should not retry.. another reason is a conflict, but since we can't  rely on github conflict resolution, I prefer to abort the script.

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:235]: removed the iterating in favor of random seed to make sure branch name is unique.

if there is some github downtime I prefer it would exit the script, don't you agree?

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:242]: Done

File: tools/bumper/cnao_repo_commands.go
Comment (RamLavi) [reply tools/bumper/cnao_repo_commands.go:243]: I found the proper way to check if the request got `not found` : https://github.com/google/go-github/blob/76c3c3d7c6e78e8c91e77d2e2578c4e0a7cf96ea/github/git_refs_test.go#L73
fixed

**Comment by RamLavi** [comment]
/hold cancel 
#619 merged, PR rebased

File: tools/bumper/cnao_repo_commands.go
Comment (qinqon) [reply tools/bumper/cnao_repo_commands.go]: agree, closing this.

**Pull request merged**
